kiwmi/desktop/output.c

```diff
@@ -381,8 +420,126 @@ new_output_notify(struct wl_listener *listener, void *data)
     wl_signal_emit(&desktop->events.new_output, output);
 }
 
+static void
+output_manager_configure(
+    struct kiwmi_server *server,
+    struct wlr_output_configuration_v1 *configuration,
+    bool test)
+{
+    struct wlr_output_configuration_head_v1 *head;
+    struct wlr_output_configuration_head_v1 *bad_head = NULL;
+    wl_list_for_each (head, &configuration->heads, link) {
+        struct wlr_output *wlr_output = head->state.output;
+
+        wlr_output_enable(wlr_output, head->state.enabled);
+        if (head->state.enabled) {
+            if (head->state.mode) {
+                wlr_output_set_mode(wlr_output, head->state.mode);
+            } else {
+                wlr_output_set_custom_mode(
+                    wlr_output,
+                    head->state.custom_mode.width,
+                    head->state.custom_mode.height,
+                    head->state.custom_mode.refresh);
+            }
+
+            wlr_output_set_transform(wlr_output, head->state.transform);
+            wlr_output_set_scale(wlr_output, head->state.scale);
+        }
+
+        if (!wlr_output_test(wlr_output)) {
+            bad_head = head;
+            break;
+        }
+    }
+
+    wl_list_for_each (head, &configuration->heads, link) {
+        if (bad_head != NULL || test) {
+            wlr_output_rollback(head->state.output);
+            if (head == bad_head) {
+                break;
+            } else {
+                continue;
+            }
```

Comment:
My logic would be, *if we’ve gotten here, `continue`*, instead of *either `break`, or `continue`*. Seems very subjective though.

```suggestion
            if (head == bad_head) {
                break;
            }
            continue;
```

Reply:
If there's jumps on both outcomes of the if, I generally like to use if/else.  Makes it clearer to me that the if statement is a for in the control flow rather than a detour that will rejoin.

I can change it if you'd prefer though.